### PR TITLE
CLD-8395 Integrate playwright into E2E tests

### PR DESCRIPTION
**Pull request by mvitale1989**
#### Summary

Integrate Playwright tests into the usual E2E testing mechanism. Overview of changes:
- Complete Playwright integration into the E2E testing framework (e.g. `TEST_FILTER` supports Playwright too)
- A full E2E test will now run both Cypress and Playwright
   * Each will result in its own status check
   * Playwright will post back to the Test Automation community channel, with its own run results
- The E2E test template Workflow returns the test results back to the parent workflow
  * Corollary: we're not able to write a run result summary in a comment too, when the E2E test run is complete
- Unify how Cypress&Playwright report their run results (`{cypress,playwright}/results/summary.json`), to keep the Workflow agnostic
- Decrease code duplication

Why this PR is important: we recommend writing new E2E tests in Playwright, but until now we had no way to run Playwright in automation. So if a developer added their tests in a PR, that wouldn't have any effect on the E2E pass/fail check mark, once we enforce E2E tests for PRs.
This is even more important as there was the recent discussion about also enforcing writing E2E tests for feature PRs.

#### Ticket Link

https://mattermost.atlassian.net/browse/CLD-8395

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by mvitale1989** [comment]
:point_up: the rolling release run looks good, triggering the normal E2E run now

File: .github/workflows/e2e-fulltests-ci.yml [hunk 1]
```diff
@@ -241,12 +242,45 @@ jobs:
...
+    strategy:
+      matrix:
+        type:
+        - name: PR
+    with:
+      commit_sha: "${{ needs.generate-test-variables.outputs.commit_sha }}"
+      status_check_context: "${{ needs.generate-test-variables.outputs.status_check_context }}-playwright"
+      workers_number: "1"
+      testcase_failure_fatal: false # NB: do not enable until Playwright test stability is reached
+      run_preflight_checks: false
```
Comment (saturninoabril) [inline .github/workflows/e2e-fulltests-ci.yml:258]: Why false?
I noticed this was set to false in Cypress too which skipped running lint. In Playwright, `npm run check` includes tsc/lint/prettier.

File: e2e-tests/playwright/package.json [hunk 2]
```diff
@@ -22,6 +22,7 @@
         "async-wait-until": "2.0.12",
         "axe-core": "4.10.0",
         "chalk": "4.1.2",
+        "dayjs": "^1.11.13",
```
Comment (saturninoabril) [inline e2e-tests/playwright/package.json:25]: nit: let's use fix version.
```suggestion
        "dayjs": "1.11.13",
```

**Review by saturninoabril — APPROVED** [review]
This is awesome. Great work and thanks @mvitale1989!

(Left non-blocking comments.)

**Review by toninis — APPROVED** [review]

File: .github/workflows/e2e-fulltests-ci.yml (hunk 1 above)
Comment (mvitale1989) [reply .github/workflows/e2e-fulltests-ci.yml:258]: This assumes that if we're running the E2E tests in full, the [smoketests](https://github.com/mattermost/mattermost/blob/master/.github/workflows/e2e-tests-ci.yml) have already been run and were successful, and they performed the linting on the commit code already for both Playwright and Cypress.

Comment (saturninoabril) [reply .github/workflows/e2e-fulltests-ci.yml:258]: Oh, right, got it, thanks!

**Pull request merged**